### PR TITLE
GCPrivate, GCHandle, and GChandlePrivate events for local GC

### PR DESCRIPTION
**Pull request by davmason**

### CODE REVIEW

File: src/gc/handletable.cpp [hunk 1]
```diff
@@ -601,8 +602,8 @@ void HndLogSetEvent(OBJECTHANDLE handle, _UNCHECKED_OBJECTREF value)
         ADIndex appDomainIndex = HndGetHandleADIndex(handle);   
         AppDomain* pAppDomain = SystemDomain::GetAppDomainAtIndex(appDomainIndex);
```
Comment (davmason) [inline src/gc/handletable.cpp]: This is the only call which I wasn't sure if it should go on the vm or the gc side. Let me know if I should move it over

Comment (swgillespie) [reply src/gc/handletable.cpp]: That should go on the EE side, I think. Ideally the GC doesn't need to know about AppDomains.

**Comment by davmason** [comment]
@swgillespie Actually, looking at this the handle table stores the appdomain information so the calls don't go in to the vm at all. It doesn't make sense to move. The vm would have to ask the handle table which appdomain it's associated with if moved, so it would need to call back in to the local gc dll.

**Comment by swgillespie** [comment]
@davmason ah, that's unfortunate. I think we can figure that out in a follow-up PR, then.

**Pull request merged**